docs/docs/03-light-clients/04-wasm/03-integration.md

```diff
@@ -0,0 +1,240 @@
+   // This sample code uses a constructor function that
+   // accepts a pointer to an existing instance of Wasm VM.
+   // This is the recommended approach when the chain
+   // also uses `x/wasm`, and then the Wasm VM instance
+   // can be shared.
+   // See the section below for more information.
+  app.WasmClientKeeper = wasmkeeper.NewKeeperWithVM(
+    appCodec,
+    keys[wasmtypes.StoreKey],
+    authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+    wasmVM,
+  )  
+  app.ModuleManager = module.NewManager(
+    // SDK app modules
+    ...
+    wasm.NewAppModule(app.WasmClientKeeper),
+  ) 
+  app.ModuleManager.SetOrderBeginBlockers(
+    ...
+    wasmtypes.ModuleName,
+    ...
+  ) 
+  app.ModuleManager.SetOrderEndBlockers(
+    ...
+    wasmtypes.ModuleName,
+    ...
+  ) 
+  genesisModuleOrder := []string{
+    ...
+    wasmtypes.ModuleName,
+    ...
+  }
+  app.ModuleManager.SetOrderInitGenesis(genesisModuleOrder...)
+  app.ModuleManager.SetOrderExportGenesis(genesisModuleOrder...)
+  ...
+
+	// initialize BaseApp
+  app.SetInitChainer(app.InitChainer)
+  ...
+
+  // must be before Loading version
+  if manager := app.SnapshotManager(); manager != nil {
+    err := manager.RegisterExtensions(
+      wasmkeeper.NewWasmSnapshotter(app.CommitMultiStore(), &app.WasmClientKeeper),
+    )
+    if err != nil {
+      panic(fmt.Errorf("failed to register snapshot extension: %s", err))
+    }
+  }
+  ...
+}
+```
+
+## Keeper instantiation
+
+When it comes to instantiating `08-wasm`'s keeper there are two recommended ways of doing it. Choosing one or the other will depend on whether the chain already integrates [`x/wasm`](https://github.com/CosmWasm/wasmd/tree/main/x/wasm) or not.
+
+### If `x/wasm` is present
+
+If the chain where the module is integrated uses `x/wasm` then we recommend that both `08-wasm` and `x/wasm` share the same Wasm VM instance. Having two separate Wasm VM instances is still possible, but care should be taken to make sure that both instances do not share the directory when the VM stores blobs and various caches, otherwise unexpected behaviour is likely to happen.
+
+In order to share the Wasm VM instance please follow the guideline below. Please note that this requires `x/wasm`v0.41 or above.
+
+- Instantiate the Wasm VM in `app.go`` with the parameters of your choice.
+- [Create an `Option` with this Wasm VM instance](https://github.com/CosmWasm/wasmd/blob/db93d7b6c7bb6f4a340d74b96a02cec885729b59/x/wasm/keeper/options.go#L21-L25).
+- Add the option created in the previous step to a slice and [pass it to the `x/wasm NewKeeper` constructor function](https://github.com/CosmWasm/wasmd/blob/db93d7b6c7bb6f4a340d74b96a02cec885729b59/x/wasm/keeper/keeper_cgo.go#L36).
+- Pass the pointer to the Wasm VM instance to `08-wasm` [NewKeeperWithVM constructor function](https://github.com/cosmos/ibc-go/blob/c95c22f45cb217d27aca2665af9ac60b0d2f3a0c/modules/light-clients/08-wasm/keeper/keeper.go#L33-L38).
```

Comment:
Is there an issue to update this commit with a tag once we have one?

Reply:
No, but it's in my todo list. 😄